conformance/lang/expressions/literals/string_literal.balt

```diff
@@ -0,0 +1,371 @@
+}
+
+Test-Case: error
+Description: Test the type of the values in the iteration sequence of a string value is string.
+Labels: foreach-stmt, string
+
+public function main() {
+    string stringVar = "abc";
+    foreach int val in stringVar { // @error values in the iteration sequence of a string value is string
+    }
+}
+
+Test-Case: output
+Description: Test concatination of strings that will result in a value looks like numeric escape.
+Labels: additive-expr, string
+
+public function main() {
+    string a = "\\" + "u{61}";
+    io:println(a); // @output \u{61}
+}
+
+Test-Case: error
+Description: Test type missmatch when assigning string literal to variables of type T, where string is not a sub type of T.
+Labels: string
+
+public function main() {
+    int a = "String"; // @error string literal cannot be assigned to a variable of int
+
+    xml b = ""; // @error string literal cannot be assigned to a variable of xml
+
+    () c = ""; // @error string literal cannot be assigned to a variable of nil
+
+    byte d = "c"; // @error string literal cannot be assigned to a variable of byte
+
+    float e = "44.5"; // @error string literal cannot be assigned to a variable of float
+}
+
+Test-Case: output
+Description: Test string literal in argument context.
+Labels: string
+
+public function main() {
+    printStringLiteral("Spec Tests"); // @output Spec Tests
+
+    printStringLiteral(")(*&^%$#><:~~!`"); // @output)(*&^%$#><:~~!`
+
+    printStringLiteral("12345\t\u{0039}"); // @output 12345   9
+
+    printStringLiteral("This code point first appeared in version 1.1 of the Unicode® Standard and belongs to the \"Basic Latin\" block which goes from 0x0 to 0x7F."); // @output This code point first appeared in version 1.1 of the Unicode® Standard and belongs to the "Basic Latin" block which goes from 0x0 to 0x7F.
+}
+
+function printStringLiteral(string str) {
+    io:println(str);
+}
+
+Test-Case: output
+Description: Test string literal in defaultable parameter context.
+Labels: defaultable-param, string
+
+public function main() {
+    printString("6564"); // @output 6564
+    printString(); // @output Default
+}
+
+function printString(string str = "Default") {
+    io:println(str);
+}
+
+Test-Case: output
+Description: Test string literal in method call expression context.
+Labels: method-call-expr, string
+
+public function main() {
+    io:println("some".startsWith("s")); // @output true
+
+    io:println("Hello World".substring(6)); // @output World
+
+    io:println("ABCD".toLowerAscii()); // @output abcd
+}
+
+Test-Case: output
+Description: Test string literal in type test expression context.
+Labels: is-expr, string
+
+public function main() {
+    io:println("Book" is string); // @output true
+
+    io:println("1234" !is string); // @output false
+
+    io:println("\t\n\r\\" is string); // @output true
+}
+
+Test-Case: output
+Description: Test string literal in return statement context.
+Labels: string
```

Comment:
Since we are specifically testing returned string literals, shall we add `return-stmt` here ?

Reply:
return is included in the minimal subset, right?

Reply:
Yes, Hence resolving